tls/s2n_handshake_io.c

```diff
@@ -844,12 +896,30 @@ static int handshake_read_io(struct s2n_connection *conn)
     uint8_t record_type;
     int isSSLv2;
 
+    /* Fill conn->in stuffer necessary for the handshake */
     GUARD(s2n_read_full_record(conn, &record_type, &isSSLv2));
 
     if (isSSLv2) {
         GUARD(s2n_handshake_handle_sslv2(conn));
     }
 
+    /* In TLS 1.3, encrypted handshake records would appear to be of record type
+     * TLS_APPLICATION_DATA. The actual record content type is found after the encryped
+     * is decrypted.
+     */
+    if (conn->actual_protocol_version >= S2N_TLS13 && record_type == TLS_APPLICATION_DATA) {
```

Comment:
How confident are we that this will be the right behavior for a version of TLS that's higher than 1.3?

Reply:
I actually do prefer that we do an exact match for TLS 1.3 but we have also introduced >= TLS13 in our codebase. I'll update this instance and add a task for keeping consistency to places we do versions checks